cypress-io/github-action
# feat: add optional job summary-title parameter

- closes #1050

This PR adds an optional parameter `summary-title` to the `cypress-io/github-action`.

If specified, then the `summary-title` replaces the default Job Summary title "Cypress Results". The title is displayed at the top of a job summary.

This is especially useful when `cypress-io/github-action` is called in more than one step in a GitHub Actions workflow, so that each job summary can be easily identified with individual summary titles showing which step it came from.

## example-chrome

`summary-title` is applied to the [example-chrome](https://github.com/cypress-io/github-action/actions/workflows/example-chrome.yml) workflow.

**BEFORE**

![example-chrome_before](https://github.com/cypress-io/github-action/assets/66998419/7016d8dd-acf1-4b04-9134-55f5915e5488)

**AFTER**

![example-chrome_after](https://github.com/cypress-io/github-action/assets/66998419/366580ef-6923-4762-a529-d6b93af6753c)

`summary-title` is here defined as "Chrome headless" in the first instance and "Chrome headed" in the second instance.

[example-chrome](https://github.com/cypress-io/github-action/actions/workflows/example-chrome.yml) is the only example workflow in this repo with more than one call of `cypress-io/github-action` in a single job.

## Documentation

- The parameter `summary-title` is added to the [README](https://github.com/cypress-io/github-action/blob/master/README.md) and some inconsistencies regarding the description of the parameter `publish-summary` are removed.

## Verification

- View the log of [example-chrome](https://github.com/cypress-io/github-action/actions/workflows/example-chrome.yml) and confirm that the Job Summary uses the customized headings "Chrome headless" in the first instance and "Chrome headed" in the second instance
- View the log of [example-basic](https://github.com/cypress-io/github-action/actions/workflows/example-basic.yml) and confirm that the Job Summary heading shows "Cypress Results"

## References

- [Supercharging GitHub Actions with Job Summaries](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/)
